sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -22,7 +22,7 @@ public class TextAnalyticsClient
         internal readonly TextAnalyticsRestClient _serviceRestClient;
         internal readonly ClientDiagnostics _clientDiagnostics;
         private readonly TextAnalyticsClientOptions _options;
-        private readonly string DefaultCognitiveScope = "https://cognitiveservices.azure.com/.default";
+        private readonly string DefaultCognitiveScope = TextAnalyticsClientOptions.GetDefaultCognitiveScope(Environment.GetEnvironmentVariable("AZURE_AUTHORITY_HOST"));
```

Comment:
This won't be a reliable way of determining the cloud. Users can set the `AuthorityHost` via code as well as via the environment variable. If the scope can't be determined via the resource URI, and the service doesn't have a challenge we might need to add something like `AuthenticationScope` to the client options.

Reply:
I have updated codes, users can set different clouds at [here](https://github.com/zedy-wj/azure-sdk-for-net/blob/TA-sov-enable/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClientOptions.cs#L20). Could you review it please? I am not sure whether this is the recommended method.